### PR TITLE
Fix pluginpath for cephfs

### PR DESCRIPTION
**Pull request by Madhu-1**
# Describe what this PR does #

make pluginpath configurable for cephfs

### CODE REVIEW

File: cmd/cephcsi.go [hunk 1]
```diff
@@ -52,6 +53,8 @@ var (
 	volumeMounter   = flag.String("volumemounter", "", "default volume mounter (possible options are 'kernel', 'fuse')")
 	mountCacheDir   = flag.String("mountcachedir", "", "mount info cache save dir")
 	metadataStorage = flag.String("metadatastorage", "", "metadata persistence method [node|k8s_configmap]")
+
```
Comment (humblec) [inline cmd/cephcsi.go]: @Madhu-1 no newline required here.

File: pkg/cephfs/driver.go [hunk 2]
```diff
@@ -93,10 +93,11 @@ func NewNodeServer(d *csicommon.CSIDriver) *NodeServer {
 
 // Run start a non-blocking grpc controller,node and identityserver for
 // ceph CSI driver which can serve multiple parallel requests
-func (fs *Driver) Run(driverName, nodeID, endpoint, volumeMounter, mountCacheDir, instanceID string, cachePersister util.CachePersister) {
+func (fs *Driver) Run(driverName, nodeID, endpoint, volumeMounter, mountCacheDir, instanceID, pluginFolder string, cachePersister util.CachePersister) {
 	klog.Infof("Driver: %v version: %v", driverName, version)
 
 	// Configuration
+	PluginFolder = pluginFolder
```
Comment (humblec) [inline pkg/cephfs/driver.go]: may be you can use a different arg name in func().. but even this is fine but one can miss the case of the letter and be confused.

**Comment by Madhu-1** [comment]
@humblec @ShyamsundarR @poornimag PTAL

**Comment by humblec** [comment]
@madhu-1 can you please add the issue Link in the PR description ?

**Comment by humblec** [comment]
@ShyamsundarR @poornimag Please review. This PR is pretty old , so ping with a timeout :)

**Comment by Madhu-1** [comment]
Fixes: https://github.com/ceph/ceph-csi/issues/458

**Comment by humblec** [comment]
As discussed in slack, I am merging this PR.

**Comment by jianglingxia** [comment]
@Madhu-1
the pr merge v1.2.0,but i use the csi driver v1.0,then i put the pr code to v1.0 version and make cephcsi,use the bin to build a image then deploy the two pod ,but create a pvc can not dynamic create pv and volume

the provisioner logs ,i check the mointors i set in configmap :
[root@node-7:/home/ubuntu]$ docker logs c44ec2b8c9fd
E0823 06:01:06.345306       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:06.346736       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:06.857286       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:06.857351       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:07.865603       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:07.865630       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:09.879447       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:09.879544       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:13.888292       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:13.888312       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:21.946793       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:21.946814       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set
E0823 06:01:37.963374       1 controllerserver.go:59] validation of volume options failed: either monitors or monValueFromSecret should be set
E0823 06:01:37.963398       1 utils.go:109] GRPC error: rpc error: code = InvalidArgument desc = either monitors or monValueFromSecret should be set

**Comment by Madhu-1** [comment]
@jianglingxia there are a lot of changes between v1.0.0 and v1.2.0. 

from the error, it looks like csi is not able to fetch mon values. btw how you are passing mon values?

**Comment by jianglingxia** [comment]
yes,the v1.0.0 and v1.2.0 has a lot of changes, the mon values i writer the configmap and create configmap

because that my kubernetes cluster version is v1.13,so then use the csi version v1.0.0,can you give my some idea?thanks

**Comment by Madhu-1** [comment]
@jianglingxia cephcsi v1.2.0 works with Kube 1.13. why you are using v1.0.0?

**Comment by Madhu-1** [comment]
@jianglingxia if you want to use v1.0.0 can you add monitors in storageclass and try?

**Comment by jianglingxia** [comment]
can i use k8s v1.13 and use csi version v1.2.0? thanks
but why before i use your images madhupr001/cephcsi:canary can right?and the storageclass do not add moniors

**Comment by Madhu-1** [comment]
yes you can  use cephcsi v1.2.0 with kube 1.13

**Comment by jianglingxia** [comment]
but 
https://github.com/ceph/ceph-csi/issues/524 the issue if use multi replicas must need k8s v1.14

the pr  https://github.com/kubernetes-csi/external-provisioner/pull/249  merge the sidecar,the k8s code change it when support multi replicas ?

**Comment by Madhu-1** [comment]
you can use a single replica.  refer templates from v1.13 folder example https://github.com/ceph/ceph-csi/tree/master/deploy/rbd/kubernetes/v1.13

**Comment by jianglingxia** [comment]
sorry i see the pr label v1.2.0, but the branch and realeas can not find v1.2.0,only v1.0.0 and v1.1.0

**Comment by Madhu-1** [comment]
yes, we have only released v1.1.0 and v1.0.0  we are yet to release v1.2.0 this month End.

if you are testing ceph-csi you can use the master branch

**Pull request merged**